internal/controller/controllers/kube_compat.go

```diff
@@ -18,3 +30,168 @@ func ServerIsOpenShift(ctx context.Context, c client.Client) (bool, error) {
 	}
 	return false, client.IgnoreNotFound(err)
 }
+
+func serverSupportsCertManager(ctx context.Context, c client.Client) (bool, error) {
+	certCRD := apiextensionsv1.CustomResourceDefinition{}
+	err := c.Get(ctx, types.NamespacedName{Name: certificateCRDName}, &certCRD)
+	if err == nil {
+		return true, nil
+	}
+	return false, client.IgnoreNotFound(err)
+}
+
+func newIngress(ctx context.Context, log logrus.FieldLogger, asc ASC, name string, host string, port int32) (client.Object, controllerutil.MutateFn, error) {
+	if asc.spec.Ingress == nil {
+		return nil, nil, fmt.Errorf("ingress config is required for non-OpenShift deployments")
+	}
+	pathTypePrefix := netv1.PathTypePrefix
+	ingress := &netv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: asc.namespace,
+		},
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(asc.Object, ingress, asc.rec.Scheme); err != nil {
+			return err
+		}
+		ingress.Spec = netv1.IngressSpec{
+			IngressClassName: asc.spec.Ingress.ClassName,
+			Rules: []netv1.IngressRule{{
+				Host: host,
+				IngressRuleValue: netv1.IngressRuleValue{HTTP: &netv1.HTTPIngressRuleValue{
+					Paths: []netv1.HTTPIngressPath{{
+						Path:     "/",
+						PathType: &pathTypePrefix,
+						Backend: netv1.IngressBackend{
+							Service: &netv1.IngressServiceBackend{
+								Name: name,
+								Port: netv1.ServiceBackendPort{
+									Number: port,
+								},
+							},
+						},
+					}},
+				}},
+			}},
+		}
+		return nil
+	}
+
+	return ingress, mutateFn, nil
+}
+
+func certManagerComponents() []component {
+	return []component{
+		{"SelfSignedIssuer", aiv1beta1.ReasonCertificateFailure, newSelfSignedIssuer},
+		{"CAIssuer", aiv1beta1.ReasonCertificateFailure, newCAIssuer},
+		{"CACert", aiv1beta1.ReasonCertificateFailure, newCACert},
+		{"WebhookCert", aiv1beta1.ReasonCertificateFailure, newWebhookCert},
+	}
+}
+
+// newSelfSignedIssuer describes how to create an Issuer for the self-signed root issuer
+func newSelfSignedIssuer(ctx context.Context, log logrus.FieldLogger, asc ASC) (client.Object, controllerutil.MutateFn, error) {
```

Comment:
very nit: can all the names be constants?
E.g. `Name: "assisted-installer-selfsigned-ca",`

Reply:
Yeah, I was debating this so I don't care much either way. I'll change it